### PR TITLE
Ideals in CommRings

### PR DESCRIPTION
**Pull request by mzeuner**
Use `FinVec` instead of `Vec` to describe finitely generated ideals in `FGIdeals.agda`, which makes the proofs much shorter.
Define the radical and prove that the radical of an ideal is again an ideal.

### CODE REVIEW

File: Cubical/Algebra/CommRing/RadicalIdeal.agda [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ private R = fst R'
+ open CommRingStr (snd R')
+ open RingTheory (CommRing→Ring R')
+ open Sum (CommRing→Ring R')
+ open CommRingTheory R'
+ open Exponentiation R'
+ open BinomialThm R'
+ open isCommIdeal
+
+ -- is there a sqrt character?
```
Comment (ecavallo) [inline Cubical/Algebra/CommRing/RadicalIdeal.agda:42]: ``\surd``!

File: Cubical/Algebra/CommRing/FGIdeal.agda [hunk 2]
```diff
@@ -12,140 +12,90 @@ open import Cubical.Foundations.Powerset
...
+                                                            ∙∙ ∑Ext λ i → ·Assoc r (α .fst i) (V i)
+
+  generatedIdeal : {n : ℕ} → FinVec R n → IdealsIn Ring
+  generatedIdeal V = makeIdeal Ring
+                               (λ x → isLinearCombination V x , isPropPropTrunc)
+                               (isLinearCombination+ V)
+                               (isLinearCombination0 V)
+                               λ r → isLinearCombinationL· V r
+
+-- bad notation?
```
Comment (mortberg) [inline Cubical/Algebra/CommRing/FGIdeal.agda:91]: What does it denote?

File: Cubical/Algebra/CommRing/RadicalIdeal.agda (hunk 1 above)
Comment (mzeuner) [reply Cubical/Algebra/CommRing/RadicalIdeal.agda:44]: ah thanks, I would've never guessed that

File: Cubical/Algebra/CommRing/FGIdeal.agda (hunk 2 above)
Comment (mzeuner) [reply Cubical/Algebra/CommRing/FGIdeal.agda:91]: It's supposed to be the ideal generated by a certain FinVec, but you have to make the Ring explicit anyways...

File: Cubical/Algebra/CommRing/RadicalIdeal.agda (hunk 1 above)
Comment (ecavallo) [reply Cubical/Algebra/CommRing/RadicalIdeal.agda:44]: I happened to know this one, but usually if I don't I look up the unicode online, paste into emacs, and then use `C-u C-x =` to find out how to write it with the Agda input method.

**Pull request merged**